### PR TITLE
Add ITTAPI source for offline

### LINKED ISSUE #47989: Offline build from julia-1.8.4-full.tar.gz failes dues to missing source for ittapi
To reproduce:
- offline build julia 1.8.4 from source with julia-1.8.4-full.tar.gz
- failing with the following log when configuring llvm
```
CMake Error at lib/ExecutionEngine/IntelJITEvents/CMakeLists.txt:20 (message):
git clone https://github.com/intel/ittapi.git failed with No such file or
directory, please clone https://github.com/intel/ittapi.git
```

Workaround:
Add `LLVM_CMAKE=-DITTAPI_SOURCE_DIR=<local checkout of ittapi>` to make flags

### PR DESCRIPTION
**Pull request by vchuravy**
Fixes #47989

@NickGao or @nalimilan ould you give this a go?

### CODE REVIEW

File: deps/Makefile [hunk 1]
```diff
@@ -213,6 +217,7 @@ distcleanall: $(addprefix distclean-, $(DEP_LIBS_ALL))
 getall: $(addprefix get-, $(DEP_LIBS_ALL))
 
 include $(SRCDIR)/csl.mk
+include $(SRCDIR)/ittapi.mk
```
Comment (inkydragon) [inline deps/Makefile:220]: missing `ittapi.mk`

**Comment by vchuravy** [comment]
@vtjnash how do I fix:

```
WARNING: using mismatched version for ittapi:
  want 
  To resolve this warning, you could try either of the following suggestions: 
  1. Run the following command: make -C deps uninstall
  2. Remove the following directory: /home/vchuravy/src/julia/usr
```

File: deps/ittapi.mk [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	$(MAKE) -C $(dir $<)
+	echo 1 > $@
+
+define ITTAPI_INSTALL
+	mkdir -p $2/$$(build_libdir)
+	mkdir -p $2/$$(build_includedir)/ittapi
+	cp -a $1/bin/libittnotify.a $2/$$(build_libdir)
+	cp -a $1/bin/libjitprofiling.a $2/$$(build_libdir)
+	cp -a $1/bin/libadvisor.a $2/$$(build_libdir)
+	cp -a $(SRCCACHE)/$(ITTAPI_SRC_DIR)/include/{ittnotify.h,ittnotify-zca.h,jitprofiling.h} $2/$$(build_includedir)/ittapi/
```
Comment (staticfloat) [inline deps/ittapi.mk:26]: Sadly, this is a `bash`-ism and you can't use it everywhere.

**Comment by nalimilan** [comment]
Thanks! I confirm it fixes the problem when building the Fedora RPM.

Should probably add a mention to THIRDPARTY.md, julia.spdx.json and doc/src/devdocs/build/build.md?

File: doc/src/devdocs/build/build.md [hunk 3]
```diff
@@ -187,6 +187,7 @@ uses are listed in [`deps/$(libname).version`](https://github.com/JuliaLang/juli
 - **[mbedtls]**              — library used for cryptography and transport layer security, used by libssh2
 - **[utf8proc]**             — a library for processing UTF-8 encoded Unicode strings.
 - **[LLVM libunwind]**       — LLVM's fork of [libunwind], a library that determines the call-chain of a program.
+- **[ITTAPI]**               - Intel's Instrumentation and Traching Technology and Just-In-Time API.
```
Comment (nalimilan) [inline doc/src/devdocs/build/build.md:190]:
```suggestion
- **[ITTAPI]**               — Intel's Instrumentation and Tracing Technology and Just-In-Time API.
```

**Review by nalimilan — APPROVED** [review]

File: deps/llvm.mk [hunk 4]
```diff
@@ -286,6 +286,11 @@ configure-llvm: $(LLVM_BUILDDIR_withtype)/build-configured
 compile-llvm: $(LLVM_BUILDDIR_withtype)/build-compiled
 fastcheck-llvm: #none
 check-llvm: $(LLVM_BUILDDIR_withtype)/build-checked
+
+ifeq ($(USE_INTEL_JITEVENTS),1)
+extract-llvm: $(SRCCACHE)/$(ITTAPI_SRC_DIR)/source-extracted
```
Comment (nalimilan) [inline deps/llvm.mk:291]: Actually I think this should be:
```
$(SRCCACHE)/$(LLVM_SRC_DIR)/source-extracted: $(SRCCACHE)/$(ITTAPI_SRC_DIR)/source-extracted
```

**Comment by nalimilan** [comment]
Sorry, I think I had incorrectly checked this with `USE_BINARYBUILDER_LLVM=1`. Now with `USE_BINARYBUILDER_LLVM=0` I get the original error from https://github.com/JuliaLang/julia/issues/47989 again (see [logs](https://kojipkgs.fedoraproject.org//work/tasks/8876/99788876/build.log)).

I think one reason is that building LLVM does not extract the ITTAPI tarball, since the `compile-llvm` target depends on `build-configured`, which in turn depends on `source-extracted`, but not on `extract-llvm`. See my suggestion above to fix this.

However, even with that change, `configure-llvm` tries to clone the git repo, despite `-DITTAPI_SOURCE_DIR=/home/milan/Dev/rpmbuild/BUILD/julia-1.9.0-rc2/deps/srccache/ittapi-0014aec56fea2f30c1374f40861e1bccdd53d0cb` pointing to the correct location of the sources. Any ideas?

**Pull request merged**